src/target/stm32l0.c

```diff
@@ -591,23 +591,24 @@ static bool stm32lx_cmd_option(target_s *const target, const int argc, const cha
 	for (size_t i = 0; i < opt_size; i += 4U) {
 		const uint32_t addr = STM32Lx_NVM_OPT_PHYS + i;
 		const uint32_t val = target_mem32_read32(target, addr);
-		tc_printf(target, "0x%08" PRIx32 ": 0x%04u 0x%04u %s\n", addr, val & 0xffffU, (val >> 16U) & 0xffffU,
-			(val & 0xffffU) == ((~val >> 16U) & 0xffffU) ? "OK" : "ERR");
+		tc_printf(target, "0x%08" PRIx32 ": 0x%04" PRIu32 " 0x%04" PRIu32 " %s\n", addr, val & 0xffffU,
+			(val >> 16U) & 0xffffU, (val & 0xffffU) == ((~val >> 16U) & 0xffffU) ? "OK" : "ERR");
 	}
 
 	const uint32_t options = target_mem32_read32(target, STM32Lx_NVM_OPTR(nvm));
 	const size_t read_protection = stm32lx_prot_level(options);
 	if (stm32lx_is_stm32l1(target)) {
 		tc_printf(target,
-			"OPTR: 0x%08" PRIx32 ", RDPRT %u, SPRMD %u, BOR %u, WDG_SW %u, nRST_STP %u, nRST_STBY %u, nBFB2 %u\n",
-			options, read_protection, (options & STM32L1_NVM_OPTR_SPRMOD) ? 1 : 0,
+			"OPTR: 0x%08" PRIx32 ", RDPRT %" PRIu32 ", SPRMD %" PRIu8 ", BOR %" PRIu32 " , WDG_SW %" PRIu8
+			", nRST_STP %" PRIu8 ", nRST_STBY %" PRIu8 ", nBFB2 %" PRIu8 "\n",
+			options, (uint32_t)read_protection, (options & STM32L1_NVM_OPTR_SPRMOD) ? 1 : 0,
 			(options >> STM32L1_NVM_OPTR_BOR_LEV_S) & STM32L1_NVM_OPTR_BOR_LEV_M,
 			(options & STM32Lx_NVM_OPTR_WDG_SW) ? 1 : 0, (options & STM32L1_NVM_OPTR_nRST_STOP) ? 1 : 0,
 			(options & STM32L1_NVM_OPTR_nRST_STDBY) ? 1 : 0, (options & STM32L1_NVM_OPTR_nBFB2) ? 1 : 0);
 	} else {
-		tc_printf(target, "OPTR: 0x%08" PRIx32 ", RDPROT %u, WPRMOD %u, WDG_SW %u, BOOT1 %u\n", options,
-			read_protection, (options & STM32L0_NVM_OPTR_WPRMOD) ? 1 : 0, (options & STM32Lx_NVM_OPTR_WDG_SW) ? 1 : 0,
-			(options & STM32L0_NVM_OPTR_BOOT1) ? 1 : 0);
+		tc_printf(target, "OPTR: 0x%08" PRIx32 ", RDPROT %" PRIu32 ", WPRMOD %" PRIu16 ", WDG_SW %u, BOOT1 %u\n",
+			options, (uint32_t)read_protection, (options & STM32L0_NVM_OPTR_WPRMOD) ? 1 : 0,
```

Comment:
Here I opted to cast read_protection because it was giving me 2 different integer sizes depending on the platform I think? size_t is define as uint32_t/uint16_t depending on the platform? Maybe it will be better to redefine it as u32 everywhere?

Reply:
[`size_t` is defined by the compiler](https://en.cppreference.com/w/c/types/size_t) and trying to redefine it would invoke UB. The correct formatting string for it is `%zu`, however newlib-nano does not support this so casting to uint32_t is fine.